rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/model/TestExecutionManager.xtend

```diff
@@ -69,14 +73,28 @@ class TestExecutionManager {
 	def private TestExecutionLog createTestExecutionLog(String teLogFileName) {
 		val log = new TestExecutionLog
 		val location = logLocationHelper.logLocation
+		val testRunBaseDir = new File(location, teLogFileName)
+		log.testStatistic = readTestStatistic(testRunBaseDir)
 		log.name = teLogFileName.testExecutionLogName
-		log.logFile = new File(location, teLogFileName)
+		log.logDir = new File(testRunBaseDir, "testrun.log")
 		return log
 	}
 
+	def TestRunStatistic readTestStatistic(File parentDir) {
+		val docBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder
+		val xml = docBuilder.parse(new File(parentDir, "testSummary.xml"))
+		val testStatistic = xml.firstChild.attributes
+		return new TestRunStatistic(testStatistic.readIntFor("tests"), testStatistic.readIntFor("failures"),
+			testStatistic.readIntFor("errors"))
+	}
+
+	def int readIntFor(NamedNodeMap map, String name) {
+		return Integer.parseInt(map.getNamedItem(name).nodeValue)
+	}
+
 	def private String getTestExecutionLogName(String teLogFileName) {
-		val sdfReader = new SimpleDateFormat("yyyy.MM.dd.-HH:mm")
-		val date = sdfReader.parse(teLogFileName.substring(3, teLogFileName.lastIndexOf(".")))
+		val sdfReader = new SimpleDateFormat(TIMESTAMP_FILE_PATTERN)
+		val date = sdfReader.parse(teLogFileName.substring(8))
```

Comment:
`substring(8)` looks like a potential future bug if the pattern is changed. Can we infer this value somehow?